test/sqllogictest/cluster_log_drop.slt

```diff
@@ -0,0 +1,81 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Ensure correct log creation and destruction on `CREATE CLUSTER`, `DROP CLUSTER` DDL statements.
+
+mode cockroach
+
+## REAL starts here
+
+# The default cluster also has log sources, thus we should have one set active at boot
+query T rowsort
+select COUNT(*) from mz_catalog.mz_sources WHERE name LIKE 'mz_peek_active_%';
+----
+1
+
+statement ok
+CREATE CLUSTER c1 REPLICAS (r1 (SIZE '1'), r2 (SIZE '2'));
+
+query T rowsort
+select COUNT(*) from mz_catalog.mz_sources WHERE name LIKE 'mz_peek_active_%';
+----
+3
+
+statement ok
+DROP CLUSTER REPLICA c1.r1;
+
+query T rowsort
+select COUNT(*) from mz_catalog.mz_sources WHERE name LIKE 'mz_peek_active_%';
+----
+2
+
+statement error
```

Comment:
All of these `statement error` directives need a regex that is a submatch of the full error. For example:
```
statement error materialized view .+ definition names 2 columns, but materialized view .+ has 1 column
```

Reply:
added!